smartsim/entity/model.py

```diff
@@ -546,7 +552,10 @@ def add_function(
         db_script = DBScript(
             name=name, script=function, device=device, devices_per_node=devices_per_node
         )
-        self._append_db_script(db_script)
+        self.append_db_script(db_script)
+
+    def __hash__(self) -> int:
+        return hash(self.name)
```

Comment:
I'm not clear on the functional impact of this change

Reply:
linter complains if we have __eq__ without having __hash__. Since eq uses name as the unique matching part, hash is implemented to match that same uniqueness constraint.